soc/arm/ambiq/apollo4x/soc.c

```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright (c) 2023 Antmicro <www.antmicro.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/arch/cpu.h>
+#include <zephyr/init.h>
+#include <zephyr/kernel.h>
+#include <zephyr/logging/log.h>
+
+#include <am_mcu_apollo.h>
+#include <am_bsp.h>
+
+#include <soc.h>
+
+LOG_MODULE_REGISTER(soc, CONFIG_SOC_LOG_LEVEL);
+
+void z_arm_platform_init(void)
+{
+	am_bsp_low_power_init();
```

Comment:
`bsp`? what does this function do? we do not allow "bsp-like" code in Zephyr (we have `boards/`)

Reply:
I've replaced the BSP function with calls to actual HAL functions:
https://github.com/zephyrproject-rtos/zephyr/blob/4aace128e0d88e660dcdc4a4473f8736927196d0/soc/arm/ambiq/apollo4x/soc.c#L19-L21